numba/targets/linalg.py

```diff
@@ -667,3 +688,300 @@ def inv(context, builder, sig, args):
+                                        types.intp,  # lda
+                                        types.CPointer(a.dtype),  # wr
+                                        types.CPointer(a.dtype),  # wi
+                                        types.CPointer(a.dtype),  # vl
+                                        types.intp,  # ldvl
+                                        types.CPointer(a.dtype),  # vr
+                                        types.intp  # ldvr
+                                        )
+
+        numba_ez_rgeev = types.ExternalFunction("numba_ez_rgeev",
+                                                numba_ez_rgeev_sig)
+
+        numba_ez_cgeev_sig = types.intc(types.char,  # kind
+                                        types.char,  # jobvl
+                                        types.char,  # jobvr
+                                        types.intp,  # n
+                                        types.CPointer(a.dtype),  # a
+                                        types.intp,  # lda
+                                        types.CPointer(a.dtype),  # w
+                                        types.CPointer(a.dtype),  # vl
+                                        types.intp,  # ldvl
+                                        types.CPointer(a.dtype),  # vr
+                                        types.intp  # ldvr
+                                        )
+
+        numba_ez_cgeev = types.ExternalFunction("numba_ez_cgeev",
+                                                numba_ez_cgeev_sig)
+
+        kind = ord(get_blas_kind(a.dtype, "eig"))
+
+        JOBVL = ord('N')
+        JOBVR = ord('V')
+
+        F_layout = a.layout == 'F'
+
+        def real_eig_impl(a):
+            n = a.shape[-1]
+            if a.shape[-2] != n:
+                msg = "Last 2 dimensions of the array must be square."
+                raise numpy.linalg.LinAlgError(msg)
+
+            if not numpy.isfinite(a).all():
+                raise numpy.linalg.LinAlgError(
+                    "Array must not contain infs or NaNs.")
+
+            if F_layout:
+                acpy = numpy.copy(a)
+            else:
+                acpy = numpy.asfortranarray(a)
+
+            ldvl = 1
+            ldvr = n
+            wr = numpy.zeros(n, dtype=a.dtype)
+            wi = numpy.zeros(n, dtype=a.dtype)
+            vl = numpy.zeros((ldvl, n), dtype=a.dtype)
+            vr = numpy.zeros((ldvr, n), dtype=a.dtype)
+
+            r = numba_ez_rgeev(kind,
+                               JOBVL,
+                               JOBVR,
+                               n,
+                               acpy.ctypes,
+                               n,
+                               wr.ctypes,
+                               wi.ctypes,
+                               vl.ctypes,
+                               ldvl,
+                               vr.ctypes,
+                               ldvr)
+
+            if numpy.any(wi):
```

Comment:
The question is really whether raising an error is helpful to the user, or detrimental... Is it better to have false positives or false negatives here?
(by the way I'd expect the answer to go in a comment, so that it doesn't get forgotten ;-))